### ISSUE TITLE
Fix false negatives for last line without trailing EOL in no-eol-whitespace

### ISSUE THREAD

**Issue by vankop** [stylelint/stylelint#4218]
> Clearly describe the bug

There is no warning on the last line, when I provide code without trailing EOL

> Which rule, if any, is the bug related to?

`no-eol-whitespace`

> What CSS is needed to reproduce the bug?

any without trailing EOL, e.g.

```css
a {
  width: 00.10em;
}\u0020\u0020
```

No error on line 3 ☝🏼

```css
a {
  width: 00.10em;
}\u0020\u0020\n
```
error on line 3 ☝🏼

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "no-eol-whitespace": [true, { "ignore": ["empty-lines"] }]
  }
}
```

> Which version of stylelint are you using?

latest on https://stylelint.io/demo

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

REPL

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

no

> What did you expect to happen?

warning without trailing EOL too

> What actually happened (e.g. what warnings or errors did you get)?

no warning

May be related to #3842 
Right now code runs` styleSearch` for EOL, so last line is skipped

**Comment by evilebottnawi** [comment]
Hm, it is valid css?

**Comment by jeddy3** [comment]
I can replicate this. The last line isn't checked.

**Comment by vankop** [comment]
I will take a look on this
